tensorly/decomposition/cmtf_als.py

```diff
@@ -0,0 +1,229 @@
+    weights lambda_ and gamma are found. This implementation only works for a coupling in the
+    first mode.
+
+    Solution is found via alternating least squares (ALS) as described in Figure 5 of
+    @article{acar2011all,
+      title={All-at-once optimization for coupled matrix and tensor factorizations},
+      author={Acar, Evrim and Kolda, Tamara G and Dunlavy, Daniel M},
+      journal={arXiv preprint arXiv:1105.3422},
+      year={2011}
+    }
+
+    Notes
+    -----
+    In the paper, the columns of the factor matrices are not normalized and therefore weights are
+    not included in the algorithm.
+
+    Parameters
+    ----------
+    tensor_3d : tl.tensor or Kruskal tensor
+        3rd order tensor X = [[A, B, C]]
+    matrix : tl.tensor or Kruskal tensor
+        matrix that is coupled with tensor in first mode: Y = [[A, V]]
+    rank : int
+        rank for CP decomposition of X
+
+    Returns
+    -------
+    tensor_3d_pred, matrix_pred : Kruskal tensors
+        tensor_3d_pred = [[lambda; A,B,C]], matrix_pred = [[gamma; A,V]]
+
+    Examples
+    --------
+    A = tl.tensor([[1, 2], [3, 4]])
+    B = tl.tensor([[1, 0], [0, 2]])
+    C = tl.tensor([[2, 0], [0, 1]])
+    V = tl.tensor([[2, 0], [0, 1]])
+    R = 2
+
+    X = (None, [A, B, C])
+    Y = (None, [A, V])
+
+    tensor_3d_pred, matrix_pred = cmtf_als_for_third_order_tensor(X, Y, R)
+
+    """
+
+    if tl.is_tensor(tensor_3d):
+        X = tensor_3d
+    else:
+        _, _ = tl.kruskal_tensor._validate_kruskal_tensor(
+            tensor_3d)  # this will fail if it isn't a valid tuple or KruskalTensor
+        X = tl.kruskal_tensor.kruskal_to_tensor(tensor_3d)
+
+    if tl.is_tensor(matrix):
+        Y = matrix
+    else:
+        _, _ = tl.kruskal_tensor._validate_kruskal_tensor(
+            matrix)  # this will fail if it isn't a valid tuple or KruskalTensor
+        Y = tl.kruskal_tensor.kruskal_to_tensor(matrix)
+
+    # initialize values
+    A, B, C = initialize_factors(X.astype(float), rank, init=init)
+    V = tl.transpose(solve_least_squares(A, Y))
+    lambda_ = tl.ones(rank)
+    gamma = tl.ones(rank)
+
+    # alternating least squares
+    # note that the order of the khatri rao product is reversed since tl.unfold has another order
+    # than assumed in paper
+    for iteration in range(10 ** 4):
+        A = tl.transpose(solve_least_squares(
+            tl.transpose(tl.concatenate((tl.dot(np.diag(lambda_), tl.transpose(khatri_rao([B, C]))),
+                                         tl.dot(np.diag(gamma), tl.transpose(V))), axis=1)),
+            tl.transpose(tl.concatenate((tl.unfold(X, 0), Y), axis=1))))
+        norm_A = tl.norm(A, axis=0)
+        A /= norm_A
+        lambda_ *= norm_A
+        gamma *= norm_A
+        B = tl.transpose(solve_least_squares(tl.dot(khatri_rao([A, C]), np.diag(lambda_)),
+                                             tl.transpose(tl.unfold(X, 1))))
+        norm_B = tl.norm(B)
```

Comment:
I believe this should be the norm along `axis=0`.

Reply:
thank you!